net/core/filter.c

```diff
@@ -5849,6 +5855,278 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_pushmap, struct xdp_buff *, ctx, struct bpf_map *, map) {
+	verify_and_lock();
+	lua_pushlightuserdata(ctx->L, map);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushmap_proto = {
+	.func		= bpf_lua_pushmap,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_CONST_MAP_PTR,
+};
+
+BPF_CALL_3(bpf_lua_pushlstring, struct xdp_buff *, ctx, const char *, str, size_t, len) {
+	verify_and_lock();
+	lua_pushlstring(ctx->L, str, len);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushlstring_proto = {
+	.func		= bpf_lua_pushlstring,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+	.arg3_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_1(bpf_lua_pushskb, struct xdp_buff *, ctx) {
+	verify_and_lock();
+	lua_pushlightuserdata(ctx->L, ctx->skb);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushskb_proto = {
+	.func		= bpf_lua_pushskb,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+};
+
+BPF_CALL_2(bpf_lua_pushstring, struct xdp_buff *, ctx, const char *, str) {
+	verify_and_lock();
+	lua_pushstring(ctx->L, str);
+	return 0;
+}
+
+static const struct bpf_func_proto bpf_lua_pushstring_proto = {
+	.func		= bpf_lua_pushstring,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_VOID,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_toboolean, struct xdp_buff *, ctx, int, index) {
+	verify_and_lock();
+	return lua_toboolean(ctx->L, index);
+}
+
+static const struct bpf_func_proto bpf_lua_toboolean_proto = {
+	.func		= bpf_lua_toboolean,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_INTEGER,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_tointeger, struct xdp_buff *, ctx, int, index) {
+	verify_and_lock();
+	return lua_tointeger(ctx->L, index);
+}
+
+static const struct bpf_func_proto bpf_lua_tointeger_proto = {
+	.func		= bpf_lua_tointeger,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_INTEGER,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_2(bpf_lua_newpacket, struct xdp_buff *, ctx, int, offset) {
+	if (offset + ctx->data < ctx->data_end) {
+		return lunpack_newpacket(ctx->L, ctx->data + offset,
+				ctx->data_end - ctx->data - offset);
+	}
+
+	return -EINVAL;
+}
+
+static const struct bpf_func_proto bpf_lua_newpacket_proto = {
+	.func		= bpf_lua_newpacket,
+	.gpl_only	= false,
+	.pkt_access	= false,
+	.ret_type	= RET_INTEGER,
+	.arg1_type	= ARG_PTR_TO_CTX,
+	.arg2_type	= ARG_ANYTHING,
+};
+
+BPF_CALL_4(bpf_lua_tostring, struct xdp_buff *, ctx, char *, str, u32, size, int, index) {
+	if (lua_isstring(ctx->L, index)) {
+		strncpy(str, lua_tostring(ctx->L, index), size);
+		return 1;
+	}
```

Comment:
let's handle the exception case first.. suggestion:
```
	if (!lua_isstring(ctx->L, index))
		return 0;

	strncpy(str, lua_tostring(ctx->L, index), size);
	return 1;
```

Reply:
@VictorNogueiraRio what's the use case, if we have none, I would simply remove this helper. At least, I would rename it to something like `bpf_lua_copystring`. Also, I believe you should pop the string out of the stack, right?